e2e/pollers.go

```diff
@@ -230,3 +235,36 @@ func waitUntilPodDeleted(ctx context.Context, kube *kubeclient, podName string)
 		return err == nil, err
 	})
 }
+
+type Poller[T any] interface {
+	PollUntilDone(ctx context.Context, options *runtime.PollUntilDoneOptions) (T, error)
+}
+
+func pollVMSSOperation[T any](ctx context.Context, vmssName string, pollerOpts *runtime.PollUntilDoneOptions, vmssOperation func() (Poller[T], error)) (*T, error) {
+	var requestError azure.RequestError
+	var vmssResp T
+
+	pollErr := wait.PollImmediateWithContext(ctx, vmssOperationPollInterval, vmssOperationPollingTimeout, func(ctx context.Context) (bool, error) {
+		poller, err := vmssOperation()
+		if err != nil {
+			if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
we should only need to check the error details as a result of the `PollUntilDone` below I believe since we added the new config to the arm client

also, in the case where it's a service error, we should be checking that requestErrror.ServiceError.Code == <retriable code>, and in that case:
`return false, nil` to have the operation be retried

Reply:
off the bat, we probably wanna retry:

`AllocationFailed`
`InternalExecutionError`
`StorageFailure/SocketException`